wsd/wopi/WopiStorage.cpp

```diff
@@ -756,13 +756,18 @@ void WopiStorage::uploadLocalFileToStorageAsync(const Authorization& auth, LockC
 
         http::Session::FinishedCallback finishedCallback =
             [this, startTime, wopiLog, filePathAnonym, uriAnonym, size, isSaveAs, isRename,
-             asyncUploadCallback, profileZone = std::move(profileZone)](
-                const std::shared_ptr<http::Session>& httpSession)
+             asyncUploadCallback, profileZone = std::move(profileZone),
+             &socketPoll](const std::shared_ptr<http::Session>& httpSession)
         {
             profileZone->end();
 
             // Retire.
-            _uploadHttpSession.reset();
+            socketPoll.addCallback(
```

Comment:
Really not a huge fan of this sort of thing. We should hold references properly on the things we own. I wonder if one of these trendy: "Don't need to copy stuff" plugins is getting rid of life-saving references up the stack somewhere ? =) but ...
Adding callbacks to de-reference stuff is liable to be far less reliable, and suck during shutdown / exception paths I think - where we most likely don't invoke the callbacks before cleanup.

Surely this is not necessary; I'd be ok with an assertion that we're in the right thread through ;-)

Reply:
Glad this got removed =)